dockstore-webservice/src/main/java/io/dockstore/webservice/Helper.java

```diff
@@ -228,6 +236,102 @@ public String getContent() {
+                LOG.info(giturl);
+                if (!m.find()) {
+                    LOG.info("Namespace and/or repository name could not be found from container's giturl");
+                    return container;
+                    // throw new WebApplicationException(HttpStatus.SC_NOT_FOUND);
+                }
+
+                String url = "https://bitbucket.org/api/1.0/repositories/" + m.group(1) + "/" + m.group(2) + "/branches";
+                Optional<String> asString = ResourceUtilities.asString(url, null, client);
+                LOG.info("RESOURCE CALL: " + url);
+                if (asString.isPresent()) {
+                    String response = asString.get();
+
+                    Gson gson = new Gson();
+                    Map<String, Object> branchMap = new HashMap<>();
+
+                    branchMap = (Map<String, Object>) gson.fromJson(response, branchMap.getClass());
+                    Set<String> branches = branchMap.keySet();
+
+                    for (String branch : branches) {
+                        LOG.info("Checking branch: " + branch);
+
+                        String content = "";
+
+                        url = "https://bitbucket.org/api/1.0/repositories/" + m.group(1) + "/" + m.group(2) + "/raw/" + branch
+                                + "/Dockstore.cwl";
+                        asString = ResourceUtilities.asString(url, null, client);
+                        LOG.info("RESOURCE CALL: " + url);
+                        if (asString.isPresent()) {
+                            LOG.info("CWL FOUND");
+                            content = asString.get();
+                        } else {
+                            LOG.info("Branch: " + branch + " has no Dockstore.cwl. Checking for dockstore.cwl.");
+
+                            url = "https://bitbucket.org/api/1.0/repositories/" + m.group(1) + "/" + m.group(2) + "/raw/" + branch
+                                    + "/dockstore.cwl";
+                            asString = ResourceUtilities.asString(url, null, client);
+                            LOG.info("RESOURCE CALL: " + url);
+                            if (asString.isPresent()) {
+                                LOG.info("CWL FOUND");
+                                content = asString.get();
+                            } else {
+                                LOG.info("Branch: " + branch + " has no dockstore.cwl");
+                            }
+                        }
+
+                        // parse the collab.cwl file to get description and author
+                        if (content != null && !content.isEmpty()) {
+                            try {
+                                YamlReader reader = new YamlReader(content);
+                                Object object = reader.read();
+                                Map map = (Map) object;
+
+                                String description = (String) map.get("description");
```

Comment:
This should be re-factored. The code for parsing CWL files should not be bitbucket specific. i.e. use the same code to parse CWL no matter what git repo it comes from.  This may be better structured as two methods to get CWL from bitbucket and github respectively, and one method to parse CWL across the board.